Add a drafty R21 release note (#464)
pxblue/doc-it: src/docs/release-notes/R21/R21.mdx

> **Important**: we will be renamed to _Brightlayer User Interface_ (BLUI) soon, but our rapport to you all will stay the same 🤗

Reviewer: ```suggestion
**Important**: this will be the final release for PX Blue. For our next release, we will be renaming our platform to _Brightlayer User Interface_ (BLUI). Stay tuned for more details. 🤗
```

### Component Libraries

-   Updated Angular library to build with Angular 12.
-   Documented all the components with JSDocs. (This is the intellisense thing)

> -   Enhanced React Native component library to allow icons passed directly into components.

Reviewer: ```suggestion
-   Added new icon library for React Native projects and enhanced / simplified icon usage options within the React Native component library.
```

> -   All PX Blue React Native component now responds to device rotation properly.

Reviewer: ```suggestion
-   All PX Blue React Native components now responds to device rotation properly.
```

Reviewer: Is this new or did we do this last release?

Author: The story is in this release.

-   Angular and React Native's Channel Value component now has the option to show / hide the space between unit and value.

> -   React Native hero component now takes ChannelValueProps to control the channel value child element.

Reviewer: I don't know if I would mention items that are this small

### Component Sticker Sheet

-   Updated Figma [Component Sticker Sheet](https://www.figma.com/community/file/852558784352181868) to include 50+ (TODO: WAITING FOR SECURE CONNECT) new device- and channel-value-related icons. You can also find all of them in our [iconography](/style/iconography) page.
-   All icons (Material Design and PX Blue icons) are now searchable by their aliases in Figma.
-   Added more system components, including: phone keyboard, phone keypad, status bar, browser window, cursor, system dialog box and time picker.
-   Added table components.

### Themes

> -   Polished, pixel-perfect Angular Material chips.
> -   Snapshot testing to detect breaking changes in themes.
> -   Enhanced showcase demo for Angular and React to display all PX Blue and Material's components in an organized manner.
> -   Themed React Native Paper components.
> -   React Native dark theme, dropped RN dark theme alt.

Reviewer: ```suggestion
-   Updated styles for Angular Material components (Chips, others?).
-   Enhanced showcase demos for Angular and React to display all PX Blue and Material components in an organized manner.
-   Simplified Dark Theming for React Native (eliminated the need for alternate dark theme and provide special pre-themed wrappers for React Native Paper components).
```

### Documentation Site

> -   Updated icon embedding instructions for React Native users.

Reviewer: I wouldn't mention a minor fix like this as its own line item

> -   Updated Angular and React design pattern code examples for: Account Menu, App Bar, Forms & Validations and Loading & Waiting.

Reviewer: Should this go under a "Design Patterns" or "Code Examples" heading rather than documentation site?

Author: Theoretically, PX Blue's users access design patterns from Doc-it. Unless this release note is written for upper management, I would leave it here.

### Re-usable Authentication Workflows

-   Angular, React and React Native authentication workflows' translations are now synced up.

> -   In all three authentication workflows, Users can now go back and adjust their previous input, if allowed by the application.

Reviewer: ```suggestion
-   In all three authentication workflows, users can now go back and adjust their previous inputs during registration.
```

> -   Added more ways to see errors on the login screen

Reviewer: Might not mention this until the React one is complete

Author: Removed

**Important**: this will be the final release for PX Blue. For our next release, we will be renaming our platform to _Brightlayer User Interface_ (BLUI). Stay tuned for more details. 🤗

### Component Libraries

-   Updated Angular library to build with Angular 12.
-   Documented all the components with JSDocs, viewable with Intellisense.
-   Added new icon library for React Native projects and enhanced / simplified [icon usage options](https://pxblue-components.github.io/react-native/?path=/info/components-documentation--icon-wrapper) within the React Native component library.
-   All PX Blue React Native components now responds to device rotation properly.
-   [Angular](https://pxblue-components.github.io/angular/?path=/info/components-channel-value--readme) and [React Native](https://pxblue-components.github.io/react-native/?path=/info/components-documentation--channel-value)'s Channel Value component now has the option to show / hide the space between unit and value.

### Component Sticker Sheet

> -   Updated Figma [Component Sticker Sheet](https://www.figma.com/community/file/1024360297793425107) to include 60+ new device- and channel-value-related icons. You can also find all of them in our [iconography](/style/iconography) page.

Author: After secure connect icons are added in. Isn't it amazing ;)

-   All icons (Material Design and PX Blue icons) are now searchable by their aliases in Figma.
-   Added more system components, including: phone keyboard, phone keypad, status bar, browser window, cursor, system dialog box and time picker.
-   Added table components.

### Themes

> -   Added a new `stroked` Angular Material chip style variant.

Author: Only these two: drawer icon color bug (https://github.com/pxblue/angular-themes/pull/26) and input font family (https://github.com/pxblue/angular-themes/pull/28)

> -   Added a new `stroked` Angular Material chip style variant.

Reviewer: I would just maybe say 'Style tweaks for Angular chips, drawer, and inputs' or something like that.

-   Enhanced showcase demos for [Angular](https://github.com/pxblue/angular-showcase-demo/tree/master) and [React](https://github.com/pxblue/react-showcase-demo) to display all PX Blue and Material components in an organized manner.
-   Simplified [Dark Theming for React Native](https://github.com/pxblue/react-native-themes/tree/master#dark-theme) (eliminated the need for alternate dark theme and provide special pre-themed wrappers for React Native Paper components).

### Documentation Site

> -   Updated Angular and React design pattern code examples for: [Account Menu](/patterns/account-menu), [App Bar](/patterns/appbar), [Forms & Validations](/patterns/forms) and [Loading & Waiting](/patterns/loading).

Author: > 
Theoretically, PX Blue's users access design patterns from Doc-it. Unless this release note is written for upper management, I would leave it here.

### Re-usable Authentication Workflows

> -   Angular, React and React Native authentication workflows' translations are now synced up.
> -   In all three authentication workflows, users can now go back and adjust their previous inputs during registration.
> -   Reordered screens in the self-registration process so accepting the EULA appears as the first step.

Reviewer: ```suggestion
-   Synchronized resource string translations across all three framework implementations.
-   Added ability for users to navigate backwards and adjust previous inputs during registration.
-   EULA acceptance screen moved to be the first step of registration processes.
```

### PX Blue CLI

> -   Updated to use the latest dependencies.

Reviewer: ```suggestion
-   Updated to build projects with the latest dependencies, resources, and templates.
```

### Miscellaneous

-   Bug fixes & minor updates.
-   Added copyright information to all our package source code.

> -   Reorganized [icons](https://github.com/pxblue/icons) and [themes](https://github.com/pxblue/themes) repository.

Reviewer: ```suggestion
-   Split the [icons](https://github.com/pxblue/icons) and [themes](https://github.com/pxblue/themes) repositories into multiple repositories for easier organization and maintenance.
```

Author: We technically combined multiple icon repositories and split the themes

Reviewer: good point. Maybe 'reorganized repository structures for icons and themes'

-   Improved code coverage with unit testing and Cypress.

> ### Package Summary

Reviewer: I like adding this section. Would it be more readable if we made the lists into a table for each framework (Package...Versions Published...Changelog)?

Author: Attempting to use tables in Docit directly results in this

Author: ![image](https://user-images.githubusercontent.com/8997218/135332127-1e14d348-d2d6-47e5-bc04-528ffa32174a.png)

Author: It would take a few moment to stylize it like Material Design, and make it mobile friendly...

I would drop it... for this release

Reviewer: yeah that's fine...you could also make a quick MDX component (using the MUI table) and pass a JSON object to populate it. No CSS required.

#### Angular / Ionic

-   @pxblue/angular-components v5.0.1, v6.0.0 ([Changelog](https://github.com/pxblue/angular-component-library/releases))
-   @pxblue/angular-themes v6.3.0 ([Changelog](https://github.com/pxblue/angular-themes/releases))
-   @pxblue/angular-auth-workflow v2.4.0 ([Changelog](https://github.com/pxblue/angular-workflows/releases))
-   @pxblue/icons v1.7.0 ([Changelog](https://github.com/pxblue/icons/releases))
-   @pxblue/icons-svg v1.7.0 ([Changelog](https://github.com/pxblue/icons/releases))
-   @pxblue/ng-progress-icons v1.2.1 ([Changelog](https://github.com/pxblue/progress-icons/releases))

#### React

-   @pxblue/react-components v5.3.1 ([Changelog](https://github.com/pxblue/react-component-library/releases))
-   @pxblue/react-auth-workflow v2.4.0 ([Changelog](https://github.com/pxblue/react-workflows/releases))

> -   @pxblue/react-auth-shared v3.5.0 ([Changelog](https://github.com/pxblue/react-auth-shared/releases))

Reviewer: Meant to also say this is a behind-the-scenes package that users don't really know about, so I would leave it off this list (and off the RN one below)

Author: So you think nobody would want to refer to this list and update their existing auth workflow packages in their projects?

Reviewer: They would never need to update the react-auth-shared package separately. It doesn't even show up in their project dependency array in package.json (they just install the react-auth-workflow or react-native-auth-workflow package and pick up the necessary version of shared-auth through its child dependencies). Basically, any updates that we make in the shared-auth package would be caused by something we are changing in one of the workflow packages and we would just tell users to update to the newest version of the workflow package (which includes the shared-auth updates).

Does that make sense? 

I think 'revealing' the package behind the curtain will just confuse them and make them think they need to track the two packages (shared-auth and workflow) separately when they do not.
